### PR TITLE
Add AWS support

### PR DESCRIPTION
**Pull request by plparent**
Added support for running Zetaforge with EKS, S3 and ECR. Anvil can run on EC2.

### CODE REVIEW

File: credentials.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+)
+
+type Oracle struct {
+	RegistryAddr string
+	RegistryUser string
+	RegistryPass string
+	ClusterIP    string
+	Token        string
+	CaCert       string
+}
```
Comment (FGRCL) [inline credentials.go:32]: image

File: credentials.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+	var client clientcmd.ClientConfig
+	if config.IsLocal {
+		client = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			clientcmd.NewDefaultClientConfigLoadingRules(),
+			&clientcmd.ConfigOverrides{},
+		)
+		return client, nil
+
+	} else {
+		switch config.Cloud.Provider {
```
Comment (FGRCL) [inline credentials.go:146]: You don't want to create an interface and initialize the proper implementation based on the cloud provider? Supposedly we'll want to support other Kubernetes providers in the future, so we'd have to expand this switch statement.

File: credentials.go (hunk 1 above)
Comment (plparent) [reply credentials.go:32]: It was already there I swear.

File: credentials.go (hunk 2 above)
Comment (plparent) [reply credentials.go:146]: Yes, but this MR is already large enough. I think we should create interfaces in a future MR.

File: credentials.go [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+}
+
+func registryAddress(config Config) string {
+	switch config.Cloud.Provider {
+	case "aws":
+		return config.Cloud.AWS.RegistryAddr
+	case "oracle":
+		return config.Cloud.Oracle.RegistryAddr
+	default:
+		return ""
```
Comment (FGRCL) [inline credentials.go:212]: the Debug provider returns `""` for the registry address?

File: translator.go [hunk 4]
```diff
@@ -236,59 +250,92 @@ func kanikoTemplate(block *zjson.Block, organization string, cfg Config) *wfv1.T
...
+				Name:      "kaniko-secret",
+				MountPath: "/root/.aws/",
+			})
+
+			volumes = append(volumes, corev1.Volume{
+				Name: "kaniko-config",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: cfg.Cloud.Registry,
```
Comment (FGRCL) [inline translator.go:287]: Shouldn't this be `cfg.Cloud.aws.[...]`?

File: translator.go [hunk 5]
```diff
@@ -236,59 +250,92 @@ func kanikoTemplate(block *zjson.Block, organization string, cfg Config) *wfv1.T
...
+			volumeMounts = append(volumeMounts, corev1.VolumeMount{
+				Name:      "kaniko-config",
+				MountPath: "/kaniko/.docker/",
+			})
+		} else if cfg.Cloud.Provider == "oracle" {
+			volumes = append(volumes, corev1.Volume{
+				Name: "kaniko-secret",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: cfg.Cloud.Registry,
```
Comment (FGRCL) [inline translator.go:302]: And here also?

File: translator.go [hunk 6]
```diff
@@ -319,16 +366,23 @@ func translate(ctx context.Context, pipeline *zjson.Pipeline, organization strin
...
+			if repo {
+				err = createRepository(ctx, getImage(&block), cfg)
+				if err != nil {
+					return &workflow, blocks, err
+				}
+			}
+
 			blockPath := filepath.Join(pipeline.Build, blockKey)
 			blocks[blockPath] = ""
 			if build || !built {
```
Comment (FGRCL) [inline translator.go:388]: To build or not to built...

File: executor.go [hunk 7]
```diff
@@ -720,7 +733,7 @@ func localExecute(pipeline *zjson.Pipeline, executionId int64, executionUuid str
 
 			jsonData, err := json.Marshal(content)
 			if err != nil {
-				fmt.Sprintf("Failed to log: %s", message)
+				log.Printf("Failed to log: %s", message)
```
Comment (jmagoon) [inline executor.go:736]: This is inside of the function that writes to the logger. Using `log.Printf` inside of this function can cause a segfault or an infinite loop

File: executor.go [hunk 8]
```diff
@@ -706,7 +719,7 @@ func localExecute(pipeline *zjson.Pipeline, executionId int64, executionUuid str
 				var jsonObj map[string]interface{}
 				err := json.Unmarshal([]byte(message), &jsonObj)
 				if err != nil {
-					fmt.Printf("Failed to log: %s", message)
+					log.Printf("Failed to log: %s", message)
```
Comment (jmagoon) [inline executor.go:722]: Needs to stay `fmt` bc it's inside the logger, already, and this can cause infinite loop of logging

**Review by jmagoon — APPROVED** [review]

File: executor.go (hunk 8 above)
Comment (plparent) [reply executor.go:722]: So when should it be fmt and when should it be log?

**Comment by jmagoon** [comment]
It should always be log except for in the function that is building and
routing the logger (this one)

File: executor.go (hunk 7 above)
Comment (plparent) [reply executor.go:736]: Sprintf doesn't do anything here. I'll change it to Printf

**Review by jmagoon — APPROVED** [review]

**Pull request merged**